src/class-theme.php

```diff
@@ -35,7 +35,7 @@
  *
  * @since 1.0.0
  */
-class Main extends Lib_Core {
```

Comment:
This is called `Main` for legacy reasons I guess, because we use the same file in the plugin boilerplate and here. I don't really see a reason why we wouldn't do this, it wouldn't even break the setup process tho I'd rather if we do this for new version 5 rather than on 4. @dingo-d @iruzevic ?

Reply:
😵

Reply:
I wait for your orders :)

Reply:
plugins could be calld `Plugin`,
themes `Theme`,
applications `App` :)

Reply:
I can't resist:
streets `Main` 🤣

Reply:
Yeah, Main was chosen as it is a 'main entry point' where we can add additional services, etc.

Also, I think we have a v5 branch that we are syncing with new frontend and backend libs IIRC.

https://github.com/infinum/eightshift-boilerplate/tree/release/5.0.0

Reply:
Isn't it too late in WordPress to start an application in a theme.
A plugin runs much earlier.

Reply:
Well it's basically the same, as we need to register hooks anyhow, no? 🙂

Reply:
![kép](https://user-images.githubusercontent.com/952007/97166884-3e1bd500-1786-11eb-986b-34b886c96a36.png)

https://codex.wordpress.org/Plugin_API/Action_Reference

not much, really

I think code's worst part is the theme.

- no custom routing
- no route to page template association
- no templating!
- no structure at all: just plain files `include`-ed (called the "Template Hierarchy")
- you have to add a bunch of tiny crap to configure your theme

Reply:
I am aware that plugins loaded hook gets called first, but when the theme is instantiated, the relevant hooks in the boilerplate will be added in the correct place. 

The point is to have theme code run after plugin so that the relevant modifications can be made beforehand 🙂